src/Symfony/Component/String/UnicodeString.php

```diff
@@ -118,7 +122,7 @@ public function equalsTo($string): bool
         normalizer_is_normalized($string, $form) ?: $string = normalizer_normalize($string, $form);
 
         if ('' !== $string && false !== $string && $this->ignoreCase) {
-            return grapheme_strlen($string) === grapheme_strlen($this->string) && 0 === grapheme_stripos($this->string, $string);
+            return \strlen($string) === \strlen($this->string) && 0 === mb_stripos($this->string, $string, 0, 'UTF-8');
```

Comment:
@nicolas-grekas I've optimized case-insensitive equalsTo in CodePointString and UnicodeString.
Please take a look, if you see any problem with them.
(Again I'm not sure if there are any graphemes with different byte length in lowercased and uppercased variant.)

Reply:
Please check https://unicode.org/faq/casemap_charprop.html it might have the answer.
I think there are cases when the byte length differs yes.

Reply:
There are the one-to-one mappings, and the special mappings listed in ftp://ftp.unicode.org/Public/UCD/latest/ucd/SpecialCasing.txt.
Some of these special mappings have different byte length.
But the special mappings are not covered by `(grapheme|mb)_stripos` anyway.
So I think it is ok to compare the byte lengths here.

Reply:
https://3v4l.org/BdUJs